Build the manual using mdbook and redirect old page

Wilfred opened the pull request:
GitHub actions changes corresponding to https://github.com/rust-lang/rust-analyzer/pull/19015

lnicola commented on .github/workflows/ci.yml:
```diff
@@ -3,6 +3,9 @@ on:
   push:
     branches:
       - src
+  schedule:
```
Do we want this? It might be confusing for the users if the manual includes features that aren't available in the stable release yet. I think we should either have nightly and stable versions, or just stable.

Veykril replied:
Yes, once a week or manually triggered as part of a release seems preferable. I wonder if we can easily setup some form cross repo triggers? (that is once a stable releases CI run finished, trigger CI here?)

lnicola replied:
Maybe let's add `workflow_dispatch` too, so we can trigger it manually.

Wilfred replied:
Neat, I didn't know you could do that. Added.

Wilfred commented:
OK, I've rebased and changed the job to run daily, as [suggested in this comment](https://github.com/rust-lang/rust-analyzer/pull/19015#issuecomment-2613831327).

I don't see an obvious best frequency to run this. Users can run rust-analyzer from rustup (Stable is every six weeks, Nightly is presumably every day for rust-analyzer commits), or from the VS Code extension (released weekly with the latest rust-analyzer), or from their Linux distro (e.g. Arch is [roughly weekly](https://gitlab.archlinux.org/archlinux/packaging/packages/rust-analyzer/-/commits/main)).

The nice thing about running more often is contributors seeing their docs fixes reach the site sooner. I suppose it depends on how often docs fixes are version-specific. https://github.com/rust-lang/rust-analyzer/commits/2025-01-27/docs/user/manual.adoc is averaging around 1 change per month, but hopefully we will see more contributions once this lands :)

I'm happy to change to whatever frequency you folks prefer, in any case.

lnicola commented on .github/workflows/ci.yml:
```diff
@@ -16,6 +19,31 @@ jobs:
+        run: cd rust-analyzer && cargo xtask codegen
+
+      - name: Run mdbook
+        run: cd rust-analyzer/docs/book && mdbook build && cp -r book _site
```
We're copying it twice (here and below).

lnicola commented on .github/workflows/ci.yml:
```diff
@@ -16,6 +19,31 @@ jobs:
+        run: cd rust-analyzer/docs/book && mdbook build && cp -r book _site
+
+      - name: Include mdbook output
+        run: cp -r rust-analyzer/docs/book/book _site
```
Let's move it instead.

lnicola commented:
Hmm, can we move it to the parent directory so we no longer need the redirect?

Wilfred commented:
Unfortunately mdbook generates an index.html, so using the parent directory mean that https://rust-analyzer.github.io/ is entirely replaced with the mdbook content I think.

Veykril commented:
No

On Sat, Feb 1, 2025, 7:17 PM Laurențiu Nicola ***@***.***>
wrote:

> @Veykril <https://github.com/Veykril> do you have anything against
> merging this before Monday?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rust-analyzer/rust-analyzer.github.io/pull/226#issuecomment-2629055855>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4VNSYJYWW2YB2QXG4MZA32NUFSNAVCNFSM6AAAAABVYQGYWSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMMRZGA2TKOBVGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
